### PR TITLE
Kay retriever

### PR DESCRIPTION
**Pull request by plv**
- **Description**: Adding retrievers for [kay.ai](https://kay.ai) and SEC filings powered by Kay and Cybersyn. Kay provides context as a service: it's an API built for RAG.
- **Issue**: N/A
- **Dependencies**: Just added a dep to the [kay](https://pypi.org/project/kay/) package
- **Tag maintainer**: @baskaryan @hwchase17 Discussed in slack
- **Twtter handle:** [@vishalrohra_](https://twitter.com/vishalrohra_)

### CODE REVIEW

File: libs/langchain/langchain/retrievers/sec_filings.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+from typing import Any, List
+
+from langchain.callbacks.manager import CallbackManagerForRetrieverRun
+from langchain.schema import BaseRetriever, Document
+
+
+class SecFilingsRetriever(BaseRetriever):
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/sec_filings.py:9]: if this is just an instance of KayAiRetriever with specific init params, would prefer not to add a separate class for it but we can keep the example in docs!

**Comment by plv** [comment]
Sorry for the delay @baskaryan. As discussed offline I left two notebooks + 1 retriever for Kay

File: libs/langchain/langchain/retrievers/kay.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    """
+
+    client: Any
+    num_contexts: int
+
+    @classmethod
+    def create(
+        cls,
+        dataset_id: str,
+        data_sources: List[str],
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/kay.py:24]: out of curiosity why not call this data_types (to match the kay arg)?

Comment (plv) [reply libs/langchain/langchain/retrievers/kay.py]: Oh, good catch. We just renamed this arg. Fixed

**Pull request merged**